### ISSUE TITLE
[Feature Request] Get the correct type inference

### ISSUE THREAD

**Issue by 19Qingfeng** [element-plus/element-plus#1795]
### Existing Component
No

### Component Name
all

### Description
How should I get inferences when I use element and typescript.
The type of the method parameter in the component is too difficult to infer.
I don't know how to combine TS with type inference, right.
Hopefully the documentation on TS support will be completed.

**Comment by JeremyWuuuuu** [comment]
Yes, that is something we have been struggling to find out, currently IDE's seemed pretty neat without introducing `ref`, if you use `Volar` or `Vetur`, they are both good on intelligent complement.

**Comment by JeremyWuuuuu** [comment]
Also, could you give us some example about what you were struggling with?

**Comment by 19Qingfeng** [comment]
> `ref`, if you use `Volar` or `Vetur`, they are both good on intelligent complement.

TS cannot infer the correct type of ELEMENT function, I can only define the interface to implement these types，And I checked the source code and only exported the type definition of the component.



For the parameters of the method in the component, I want to know how I should correctly let the editor assume the type and pass the TS compilation stage.

``` vue
...
<el-table
     @sort-change="handleSortChange"
 />

// ...
 const handleSortChange = ({ prop, order }: { prop: string; order: string }) => {
      sortProp.value = prop;
      sortOrder.value = order;
      // fetchTableData();
 };


```

Currently I use it like this, maybe a better way can be achieved
